tests/zfs-tests/tests/functional/cli_root/zdb/zdb_objset_id.ksh

```diff
@@ -0,0 +1,78 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2019 by Datto, Inc. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# Description:
+# zdb -d dataset <objset id> will display the dataset
+#
+# Strategy:
+# 1. Create a pool
+# 2. Write some data to a file
+# 3. Get the inode number (object number) of the file
+# 4. Run zdb -d to get the objset ID of the dataset
+# 5. Run zdb <pool> -dddddd dataset objsetID objectID (decimal)
+# 6. Confirm names
+# 7. Run zdb <pool> -dddddd dataset objsetID objectID (hex) 
+# 8. Confirm names
```

Comment:
Another useful way to test this, and I suspect a common use case would be to get the objset ID from the `/proc/spl/kstat/zfs/[pool]/objset-[objset-id]` and verify that running `zdb` with the objset ID matches the `dataset_name` from proc.

Reply:
Added this to test script.

Reply:
We do not have an a way to do this on FreeBSD. Can this be done in a separate test and added to linux.run?

Reply:
I think wrapping this additional extra sanity check with `is_linux` would be fine.

Reply:
That works :+1: